### PR TITLE
ORV2 1901  TPS Monitoring For Stuck Records

### PR DESCRIPTION
**Pull request by gchauhan-aot**
# Description

TPS Monitoring For Stuck Records

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A Tested Manually
- [ ] Test B

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1329-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1329-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1329-dops.apps.silver.devops.gov.bc.ca/api)
- [TPS-Migration](https://onroutebc-1329-tps-migration.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

### CODE REVIEW

File: tps-migration/src/modules/tps-permit/tps-permit.service.ts [hunk 1]
```diff
@@ -137,88 +135,184 @@ export class TpsPermitService {
...
+   * This method identifies stuck permits and verifies if their corresponding documents have been uploaded.
+   * If a document already exists, the process takes no further action.
+   * Otherwise, it attempts to upload the permit document again to S3 storage and updates the reference in the ORBC permit table.
+   */
+  @Cron(`${process.env.TPS_MONITORING_POLLING_INTERVAL || '0 0 1 * * *'}`)
+  @LogAsyncMethodExecution()
+  async processTpsStuckRecords() {
+    const today = new Date();
+    const yesterday = new Date(today);
+    yesterday.setDate(today.getDate() - 1);
```
Comment (john-fletcher-aot) [inline tps-migration/src/modules/tps-permit/tps-permit.service.ts:286]: What will happen if today.getDate() == 1? Will this still work supplying 0 to setDate()?

Comment (john-fletcher-aot) [reply tps-migration/src/modules/tps-permit/tps-permit.service.ts:286]: Alternately, if today.getDate() == 30 and prev month (Feb) has only 28 days

Comment (gchauhan-aot) [reply tps-migration/src/modules/tps-permit/tps-permit.service.ts:286]: day =1 sets the date to the last day of previous month

Comment (john-fletcher-aot) [reply tps-migration/src/modules/tps-permit/tps-permit.service.ts:286]: second comment is dumb, first comment is handled automatically by JS by setting day to last day of prev month when it gets 0 passed in.

**Review by john-fletcher-aot — APPROVED** [review]

**Pull request merged**